ui/shared/ad/useCustomBanners.tsx

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from '@tanstack/react-query';
+import React from 'react';
+
+import type { AdCustomConfig } from 'types/client/adCustomConfig';
+
+import config from 'configs/app';
+import type { ResourceError } from 'lib/api/resources';
+import useApiFetch from 'lib/hooks/useFetch';
+
+const feature = config.features.adsBanner;
+const configUrl = (feature.isEnabled && feature.provider === 'custom') ? feature.configUrl : '';
+export default function useCustomBanners() {
+  const apiFetch = useApiFetch();
+  const { isLoading, data } = useQuery<unknown, ResourceError<unknown>, Array<AdCustomConfig>>(
+    [ 'custom-configs' ],
+    async() => apiFetch(configUrl),
+    {
+      enabled: feature.isEnabled && feature.provider === 'custom',
+      staleTime: Infinity,
+    });
+
+  return React.useMemo(() => ({
```

Comment:
I assume that result from `'@tanstack/react-query'` is already memoized. So, this is unnecessary. You can just return `useQuery` call result from here. And probably this hook doesn't deserve to be in a separate file, imho.

Reply:
Okay, I will optimize this section.